fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java

```diff
@@ -0,0 +1,108 @@
+    /**
+     * Get Iceberg table identifier by table property
+     */
+    public static TableIdentifier getIcebergTableIdentifier(IcebergTable table) {
+        return TableIdentifier.of(table.getDb(), table.getTable());
+    }
+
+    /**
+     * Get Iceberg table identifier by table property
+     */
+    public static TableIdentifier getIcebergTableIdentifier(String db, String table) {
+        return TableIdentifier.of(db, table);
+    }
+
+    /**
+     * Returns the corresponding catalog implementation.
+     */
+    public static IcebergCatalog getIcebergCatalog(IcebergTable table)
+            throws StarRocksIcebergException {
+        IcebergCatalogType catalogType = table.getCatalogType();
+        return getIcebergCatalog(catalogType, table.getIcebergHiveMetastoreUris());
+    }
+
+    /**
+     * Returns the corresponding catalog implementation.
+     */
+    public static IcebergCatalog getIcebergCatalog(IcebergCatalogType catalogType, String metastoreUris)
+            throws StarRocksIcebergException {
+        switch (catalogType) {
+            case HIVE_CATALOG: return IcebergHiveCatalog.getInstance(metastoreUris);
+            default:
+                throw new StarRocksIcebergException(
+                        "Unexpected catalog type: " + catalogType.toString());
+        }
+    }
+
+    /**
+     * Get hdfs file format in StarRocks use iceberg file format.
+     * @param format
+     * @return HdfsFileFormat
+     */
+    public static HdfsFileFormat getHdfsFileFormat(FileFormat format) {
+        switch (format) {
+            case ORC:
+                return HdfsFileFormat.ORC;
+            case PARQUET:
+                return HdfsFileFormat.PARQUET;
+            default:
+                throw new StarRocksIcebergException(
+                        "Unexpected file format: " + format.toString());
+        }
+    }
+
+    /**
+     * Get current snapshot of iceberg table, return null if snapshot do not exist.
+     * Refresh table is needed.
+     * @param table
+     * @return Optional<Snapshot>
+     */
+    public static Optional<Snapshot> getCurrentTableSnapshot(Table table, boolean refresh) {
+        if (refresh) {
+            table.refresh();
+        }
+        return Optional.ofNullable(table.currentSnapshot());
+    }
+
+    /**
+     * Get table scan for given table and snapshot, filter with given iceberg predicates.
+     * Refresh table if needed.
+     * @param table
+     * @param snapshot
+     * @param icebergPredicates
+     * @param refresh
+     * @return
+     */
+    public static TableScan getTableScan(Table table,
+                                         Snapshot snapshot,
+                                         List<UnboundPredicate> icebergPredicates,
+                                         boolean refresh) {
+        if (refresh) {
+            table.refresh();
+        }
+        AtomicReference<TableScan> tableScan = new AtomicReference<>(table.newScan()
```

Comment:
Why do we need the advanced `AtomicReference` here ?  In our flink/spark code, we use a simple for-loop to update the scan for each filter: https://github.com/apache/iceberg/blob/1d8164383baf0e343ade13b21ed0e4d4d02c56cd/flink/v1.13/flink/src/main/java/org/apache/iceberg/flink/source/FlinkSplitGenerator.java#L81